adapters/ion-adapter/src/test/java/com/connexta/replication/adapters/ion/IonNodeAdapterTest.java

```diff
@@ -0,0 +1,171 @@
+package com.connexta.replication.adapters.ion;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.method;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+
+import com.google.common.collect.ImmutableList;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+import java.util.Collections;
+import javax.annotation.Nullable;
+import org.codice.ditto.replication.api.data.CreateStorageRequest;
+import org.codice.ditto.replication.api.data.Metadata;
+import org.codice.ditto.replication.api.data.Resource;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.test.web.client.response.DefaultResponseCreator;
+import org.springframework.test.web.client.response.MockRestResponseCreators;
+import org.springframework.web.client.RestTemplate;
+
+public class IonNodeAdapterTest {
+
+  RestTemplate restTemplate;
+
+  MockRestServiceServer mockServer;
+
+  IonNodeAdapter adapter;
+
+  @Before
+  public void setup() throws Exception {
+    restTemplate = new RestTemplate();
+    mockServer = MockRestServiceServer.createServer(restTemplate);
+    adapter = new IonNodeAdapter(new URL("http://localhost:1234"), restTemplate);
+  }
+
+  @Test
+  public void isAvailable() {
+    DefaultResponseCreator response = MockRestResponseCreators.withSuccess();
+    HttpHeaders headers = new HttpHeaders();
+    headers.addAll("Allow", ImmutableList.of("POST", "OPTIONS"));
+    response.headers(headers);
+    mockServer
+        .expect(requestTo("http://localhost:1234/ingest"))
+        .andExpect(method(HttpMethod.OPTIONS))
+        .andRespond(response);
+    assertThat(adapter.isAvailable(), is(true));
+    mockServer.verify();
+  }
+
+  @Test
+  public void isNotAvailable() throws Exception {
+    restTemplate = new RestTemplate();
+    adapter = new IonNodeAdapter(new URL("http://localhost:1234321"), restTemplate);
+    assertThat(adapter.isAvailable(), is(false));
+  }
+
+  @Test
+  public void getSystemName() {
+    assertThat(adapter.getSystemName(), is("ION"));
+  }
+
+  @Test
+  public void exists() {
+    // we always return false since we don't have a way to check right now
+    assertThat(adapter.exists(null), is(false));
```

Comment:
Do we even want this test? What value does it provide? The coverage numbers are pretty high without it. Same question for getSystemName on line 67.

Reply:
Removed this test but left the system name one.